homeassistant/components/broadlink/switch.py

```diff
@@ -47,6 +48,7 @@
         vol.Required(CONF_NAME): cv.string,
         vol.Optional(CONF_COMMAND_OFF): data_packet,
         vol.Optional(CONF_COMMAND_ON): data_packet,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
```

Comment:
Per [ADR-0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md), we no longer allow integrations to be configured via YAML – they must be configured via the UI. We allow existing YAML configuration to persist, but not alter that configuration.

Reply:
In Home Assistant 2022.10 unique ID has been added for universal_media_player. What is different hère? 😔

Reply:
I did not review that PR, nor am I a user of `universal_media_player`, but after perusing its docs, I guess that a special exception was made because that platform can't be converted to a purely UI-based setup straightforwardly. [From the "Decision" heading in ADR-0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision):

> Integrations that communicate with devices and/or services are only configured via the UI. In rare cases, we can make an exception.

I don't think that applies here, but I'll mark this PR as needing additional review.

Reply:
It looks like this platform reimplements template switches

https://www.home-assistant.io/integrations/broadlink/#setting-up-custom-irrf-switches

It should probably be deprecated and removed instead. Template switches already allow setting a unique id.
https://www.home-assistant.io/integrations/switch.template/